react-immutable-state/react-immutable-state_car-race/components/CarRace/index.js

```diff
@@ -8,13 +8,39 @@ import {
   Track,
 } from "./CarRace.styled";
 import { initialCars, getRandomDistance } from "../../utils/utils";
+import { useImmer } from "use-immer";
 
 const finishLine = 200;
 
 export default function CarRace() {
-  const [cars, setCars] = useState(initialCars);
+  //const [cars, setCars] = useState(initialCars);
+  const [cars, updateCars] = useImmer(initialCars);
 
   function moveCar(clickedCar) {
+    /*setCars((cars) =>
+      cars.map((car) =>
+        // no id!
+        car.emoji === clickedCar.emoji
+          ? {
+              ...car,
+              position: {
+                // copy data first, change afterwards
+                ...car.position,
+                x: car.position.x + coveredDistance,
+                lastDistance: coveredDistance,
+              },
+            }
+          : car
+      )
+    );*/
+
+    updateCars((draft) => {
+      const car = draft.find((car) => car.emoji === clickedCar.emoji);
+
+      car.position.x = car.position.x + coveredDistance;
+      car.position.lastDistance = coveredDistance;
```

Comment:
`car.position.lastDistance += coveredDistance`  `+=`  Otherwise, the shadow of the car will not update correctly